Added read_entity_state to DurableOrchestrationClient

martijnlentink opened the pull request:
I couldn't find any reason why this function wasn't implemented in Python. I'm using this for my own project right now, should we add this?

davidmrdavid reviewed (commented):
Thank you for this contribution, really appreciate it! I don't see anything wrong with the implementation, but could you please add a _sample_ for this new API (in the `samples` folder) so that users can have an example on how to use this? It would also make it easier for us to validate that it's working as expected. Thanks!

martijnlentink commented:
I added a function to the Counter sample and updated some of the comments to comply with the `flake8-docstrings`. Let me know if you need anything else 👍

davidmrdavid commented:
It looks great! I'll get back to you after validating this locally but it looks great to me, thank you ⚡ ⚡

davidmrdavid commented on samples/counter_entity/RetrieveEntity/function.json:
```diff
@@ -0,0 +1,25 @@
+      "name": "req",
+      "type": "httpTrigger",
+      "direction": "in",
+      "route": "entity/{entityName:alpha}/{entityKey}",
```
Why do we need the `:alpha` in the route?

davidmrdavid replied:
Why do we need the `:alpha` in the route?

davidmrdavid commented on samples/counter_entity/RetrieveEntity/function.json:
```diff
@@ -0,0 +1,25 @@
+      "name": "req",
+      "type": "httpTrigger",
+      "direction": "in",
+      "route": "entity/{entityName:alpha}/{entityKey}",
```
Why do we need the :alpha in the route?

martijnlentink replied:
Good point, we don't actually *need* it. I'll remove it to avoid any confusion.

davidmrdavid reviewed (commented):
Finally got around to validating this, and it works great! Thank you so much for this PR. I'm almost ready to approve this, but I have one question I need answered before doing so. Please see my question below. Thanks!

davidmrdavid reviewed (approved):
LGTM! Thank you for this contribution 🎉
